### DISCUSSION TITLE
NextJS 13 - Redirect Error

### DISCUSSION THREAD

**Discussion by kevinswarner** [vercel/next.js#42093] [Help]
I am using the redirect function within my server component, and I notice that the browser shows an error in the console. The redirect works, but the error is there.

The error is... Uncaught Error: NEXT_REDIRECT

So I checked the new documentation...

https://beta.nextjs.org/docs/api-reference/redirect

And it says the redirect function throws an error. What I don't understand is why it throws an error. The error pollutes the console in the browser and seems strange.

Does anyone know why this is done in this way? Thanks!

**Comment by OMikkel** [comment]
I am asking the same question, an error is something that needs attention and does not work.
Maybe it is a way of terminating the server component so it does not keep rendering, but there must be a more elegant way instead of logging an error.

![](https://i.imgur.com/dwMHwV7.png)
